### PR TITLE
Improve API consistency

### LINKED ISSUE #83: Make API consistent
We're using the builder pattern in some cases:
- Eip712TransactionRequest
- DepositRequest

We need to add consistency across the board. For example, create a `WithdrawalRequest` in similar fashion.

As part of this taks we should also look into other potential consistency issues.

### PR DESCRIPTION
**Pull request by IAvecilla**
closes #83

### CODE REVIEW

**Review by mationorato — APPROVED** [review]
LGTM

**Review by jpcenteno — APPROVED** [review]
Maybe we could rename the `XyzRequest` structs to `XyzBuilders` to be more obvious to new SDK users.

Overall, it looks good to me. I think that all my comments can be discussed and addressed in the future.

File: examples/simple_payment/main.rs
[the remark this answers is not in the document]
Comment (IAvecilla) [reply examples/simple_payment/main.rs:1]: No, we are not at the moment

File: src/eip712/transaction_request.rs [hunk 1]
```diff
@@ -229,3 +235,98 @@ impl Default for Eip712TransactionRequest {
         }
     }
 }
+
+impl TryFrom<WithdrawRequest> for Eip712TransactionRequest {
```
[the remark this answers is not in the document]
Comment (IAvecilla) [reply src/eip712/transaction_request.rs:239]: The `TryFrom` function already implements the functionality of `TryInto`, but the reverse isn't necessarily true. By implementing `TryFrom` or `From`, you automatically gain the corresponding `TryInto` or `Into` trait respectively. This behavior is outlined in the [Rust documentation on this trait](https://doc.rust-lang.org/std/convert/trait.TryFrom.html#generic-implementations).

File: src/zks_provider/mod.rs [hunk 2]
```diff
@@ -785,14 +834,21 @@ impl<P: JsonRpcClient> ZKSProvider for Provider<P> {
...
+        tx_hash: TxHash,
         polling_time_in_seconds: Option<Duration>,
         timeout_in_seconds: Option<Duration>,
     ) -> Result<TransactionReceipt, ProviderError> {
         let polling_time_in_seconds = polling_time_in_seconds.unwrap_or(Duration::from_secs(2));
         let mut timer = tokio::time::interval(polling_time_in_seconds);
         let start = Instant::now();
 
+        let transaction_receipt =
+            self.get_transaction_receipt(tx_hash)
```
[the remark this answers is not in the document]
Comment (IAvecilla) [reply src/zks_provider/mod.rs:846]: I'm not sure that I understand the suggestion, you mean that the hash should implements the `Try_into` to a `TransactionReceipt`?

File: src/zks_wallet/wallet.rs [hunk 3]
```diff
@@ -178,62 +169,40 @@ where
...
-                    .into(),
-            )
-            .await?;
+        let transaction_receipt = era_provider
+            .send_transaction_eip712(&self.l2_wallet, request.clone())
+            .await?
+            .await?
+            .ok_or(ZKSWalletError::CustomError(
+                "No transaction receipt".to_owned(),
+            ))?;
```
[the remark this answers is not in the document]
Comment (IAvecilla) [reply src/zks_wallet/wallet.rs:200]: Seems like a good refactor for the future since this pattern is used a lot in all the SDK

**Pull request merged**